pkg/linear-pools/contracts/silo-linear-pool/SiloExchangeRateModel.sol

```diff
@@ -33,16 +33,18 @@ contract SiloExchangeRateModel {
      * @dev This function is similar to _accrueInterest function in the Silo's BaseSilo.sol contract
      * which is used to update state data that is necessary
      */
-    function calculateExchangeValue(IShareToken shareToken) external view returns (uint256) {
-        uint256 rcomp = _getCompoundInterestRate(shareToken.silo(), shareToken.asset());
-        ISilo.AssetStorage memory assetStorage = _getAssetStorage(shareToken.silo(), shareToken.asset());
+    function _calculateExchangeValue(IShareToken shareToken) internal view returns (uint256) {
+        ISilo silo = shareToken.silo();
+        address underlyingAsset = shareToken.asset();
+        uint256 rcomp = _getCompoundInterestRate(silo, underlyingAsset);
+        ISilo.AssetStorage memory assetStorage = _getAssetStorage(silo, underlyingAsset);
         uint256 accruedInterest = assetStorage.totalBorrowAmount.mulDown(rcomp);
-        uint256 protocolShareFee = _getProtocolShareFee(shareToken.silo());
+        uint256 protocolShareFee = _getProtocolShareFee(silo);
 
         uint256 protocolShare = accruedInterest.mulDown(protocolShareFee);
         // interestData.protocolFees + protocolShare = to newProtocolFees
         // Cut variable in order to be able to compile
-        ISilo.AssetInterestData memory interestData = _getInterestData(shareToken.silo(), shareToken.asset());
+        ISilo.AssetInterestData memory interestData = _getInterestData(silo, underlyingAsset);
         if (interestData.protocolFees + protocolShare < interestData.protocolFees) {
```

Comment:
In which case will this if be triggered?

Reply:
if interestData.protocolFees + protocolShare overflows. The sum will therefore be less than interestData.protocolFees

Reply:
Does this logic come directly from Silo? I'm struggling to wrap my head around why this would ever overflow during the course of normal operation, and why this particular value is the correct one to choose in the event of overflow.

Reply:
Yes it comes directly from Silo's accrueInterest function.

Reply:
@baileyspraggins Do you have a point of contact with Silo? I would love to learn more about this behavior. It seems so odd to me.

Reply:
@rabmarut Yes a guy named Edd Shadow. Do you want me to add you to the telegram?